packages/content/src/content.ts

```diff
@@ -1,8 +1,292 @@
 /* Copyright (c) 2018 Environmental Systems Research Institute, Inc.
  * Apache-2.0 */
 
+import { ISpatialReference, IExtent } from "@esri/arcgis-rest-types";
+import { IItem, getItem } from "@esri/arcgis-rest-portal";
 import { request } from "@esri/arcgis-rest-request";
+import {
+  HubType, IHubContent, IHubGeography, IHubRequestOptions, IBBox,
+  getType, getCategory, getItemThumbnailUrl, getPortalApiUrl
+} from "@esri/hub-common";
 
+function buildUrl (base: string, path: string): string {
+  const baseUrl = base.endsWith('/') ? base : `${base}/`;
+  const url = new URL(path, baseUrl);
+  return url.toString();
+}
+
+// TODO: make this real; use request() under the hood?
+export function hubRequest (url: string, requestOptions?: IHubRequestOptions) {
+  return fetch(url, {
+    method: 'POST', // TODO: get from requestOptions?
+    headers: {
+      // TODO: base on request requestOptions?
+      'Content-Type': 'application/json'
+    },
+    // TODO: base on requestOptions.params?
+    // body: JSON.stringify(body)
+  }).then(resp => {
+    if (resp.ok) {
+      return resp.json();
+    } else {
+      throw resp.statusText;
```

Comment:
i'm not a huge fan of throwing strings.

Reply:
is there an error/response type we can use consistently that includes both code and strings? @mikeringrose @tomwayson

Reply:
This now throws `RemoteServerError`, which I've copied over from the downloads package and added a chore to remove the duplicate next breaking change to #305